File: Algorithm/baekjoon/java/BJ1744.java
package baekjoon.java;

import java.util.*;
class BJ1744 {
    static long getMax(int[] inputs){
        List<Integer> plus = new ArrayList<>();
        List<Integer> minus = new ArrayList<>();
        int zero = 0;
        int one = 0;


        for(int i = 0; i < inputs.length; ++i){
            if(inputs[i] == 1) one += 1;
            else if(inputs[i] > 0) plus.add(inputs[i]);
            else if(inputs[i] < 0) minus.add(inputs[i]);
            else zero += 1;
        }

        long output = one;

        Collections.sort(plus);
        Collections.reverse(plus);
        if(plus.size() % 2 == 1){
            plus.add(1);
        }

        for(int i = 0; i < plus.size() ; i += 2){
            output += plus.get(i) * plus.get(i + 1);
        }

        Collections.sort(minus);

        if(minus.size() % 2 == 1){
           if(zero > 0) minus.add(0);
           else minus.add(1);
        }

        for(int i = 0; i < minus.size(); i += 2){
            output += minus.get(i) * minus.get(i + 1);
        }

        return output;
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] inputs = new int[n];
        for(int i = 0; i < n; ++i){
            inputs[i] = sc.nextInt();
        }

        long output = getMax(inputs);
        System.out.println(output);
    }
}

File: Algorithm/baekjoon/swift/BJ1744.swift
func getMax(inputs: [Int]) -> Int{
    var plus : [Int] = []
    var minus : [Int] = []
    var zero = 0
    var one = 0
    
    for i in 0..<inputs.count{
        if inputs[i] == 1 {
            one += 1
        }else if inputs[i] > 0{
            plus.append(inputs[i])
        }else if inputs[i] < 0{
            minus.append(inputs[i])
        }else{
            zero += 1
        }
    }
    var output = one
    
    if plus.count % 2 == 1{
        plus.append(1)
    }
    var plusValues = plus.sorted()
    for i in stride(from: 0, to: plusValues.count, by: 2){
        output += plusValues[i] * plusValues[i+1]
    }
    
    if minus.count % 2 == 1{
        if zero > 0{
            minus.append(0)
        }else{
            minus.append(1)
        }
    }
    
    var minusValues = minus.sorted()
    for i in stride(from: 0, to: minusValues.count, by: 2){
        output += minusValues[i] * minusValues[i+1]
    }
    
    return output
}
var n = readLine()
if n != nil{
    let size = Int(n!)!
    var inputs : [Int] = []
    for _ in 0..<size{
        let input = Int(readLine()!)!
        inputs.append(input);
    }
    
    let output = getMax(inputs: inputs)
    print(output)
}
